drivers/ethernet/eth_stellaris.c

```diff
@@ -0,0 +1,365 @@
+/* STELLARIS Ethernet Controller
+ *
+ * Copyright (c) 2018 Zilogic Systems
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "dev/eth_stellaris"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ETHERNET_LEVEL
+
+#include <logging/sys_log.h>
+#include <net/ethernet.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <device.h>
+#include <soc.h>
+#include <ethernet/eth_stats.h>
+#include "eth_stellaris_priv.h"
+
+static void eth_stellaris_assign_mac(struct device *dev)
+{
+	u32_t value = 0x0;
+
+	value |= CONFIG_ETH_MAC_ADDR_0;
+	value |= CONFIG_ETH_MAC_ADDR_1 << 8;
+	value |= CONFIG_ETH_MAC_ADDR_2 << 16;
+	value |= CONFIG_ETH_MAC_ADDR_3 << 24;
+	sys_write32(value, REG_MACIA0);
+
+	value = 0x0;
+	value |= CONFIG_ETH_MAC_ADDR_4;
+	value |= CONFIG_ETH_MAC_ADDR_5 << 8;
+	sys_write32(value, REG_MACIA1);
+}
+
+static int eth_stellaris_dev_init(struct device *dev)
+{
+	u32_t value;
+
+	ARG_UNUSED(dev);
+
+	/* 1. Assign MAC address to Hardware */
+	eth_stellaris_assign_mac(dev);
+
+	/* 2. Program MCRCTL to clear RXFIFO */
+	value = BIT_MACRCTL_RSTFIFO;
+	sys_write32(value, REG_MACRCTL);
+
+	/* 3. Enable transmitter */
+	value = BIT_MACTCTL_DUPLEX | BIT_MACTCTL_CRC |
+		BIT_MACTCTL_PADEN | BIT_MACTCTL_TXEN;
+	sys_write32(value, REG_MACTCTL);
+
+	/* 4. Enable Receiver */
+	value = BIT_MACRCTL_BADCRC | BIT_MACRCTL_RXEN;
+	sys_write32(value, REG_MACRCTL);
+
+	return 0;
+}
+
+static void eth_stellaris_flush(struct device *dev)
+{
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+
+	if (dev_data->tx_pos != 0) {
+		sys_write32(dev_data->tx_word, REG_MACDATA);
+		dev_data->tx_pos = 0;
+		dev_data->tx_word = 0;
+	}
+}
+
+static void eth_stellaris_send_byte(struct device *dev, u8_t byte)
+{
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+
+	dev_data->tx_word |= byte << (dev_data->tx_pos * 8);
+	dev_data->tx_pos++;
+	if (dev_data->tx_pos == 4) {
+		sys_write32(dev_data->tx_word, REG_MACDATA);
+		dev_data->tx_pos = 0;
+		dev_data->tx_word = 0;
+	}
+}
+
+static int eth_stellaris_send(struct net_if *iface, struct net_pkt *pkt)
+{
+	struct device *dev = net_if_get_device(iface);
+	struct eth_stellaris_runtime *dev_data = DEV_DATA(dev);
+	struct net_buf *frag;
+	u16_t head_len_left, i, data_len;
+	u8_t *eth_hdr;
+	struct net_eth_hdr *pkt_hdr;
+
+	if (!pkt->frags) {
+		SYS_LOG_ERR("No data to send");
```

Comment:
Search and replace `s/SYS_//` in this file.

Reply:
Done.